### PR TITLE
Login with Oauth via Socialite

### PR DESCRIPTION
**Pull request by ferpetrelli**
This PR implements Oauth login and account creation/linking via Socialite.

To enable it you have to add to your twill configuration:

```php
'enabled' => [
    // Options
    'oauth-login' => true,
],
'oauth' => [
    'providers' => ['github', 'facebook', 'google']
],
```

Possible values for oauth providers are the ones implemented by Socialite: facebook, twitter, linkedin, google, github, gitlab and bitbucket.
See https://laravel.com/docs/6.x/socialite.

Callbacks are automatically setup for you to add them to your apps setup:
https://github.com/ferpetrelli/twill/blob/oauth/config/services.php

And credentials will be managed through environment variables:

```
GOOGLE_CLIENT_ID=123456
GOOGLE_CLIENT_SECRET=123456
FACEBOOK_CLIENT_ID=123456
FACEBOOK_CLIENT_SECRET=123456
```

### CODE REVIEW

File: src/TwillServiceProvider.php [hunk 1]
```diff
@@ -204,6 +206,14 @@ private function mergeConfigs()
      */
     private function publishMigrations()
     {
+        // New Process to publish migrations. All of them should be included
+
+        if (config('twill.enabled.oauth-login')) {
+            $this->loadMigrationsFrom(__DIR__ . '/../migrations/oauth');
```
Comment (ferpetrelli) [inline src/TwillServiceProvider.php]: @ifox We should move all migrations towards this approach for 2.0

**Comment by ifox** [comment]
Beautiful PR @ferpetrelli, thank you so much!

And definitely agree. about the new migrations approach, see https://github.com/area17/twill/pull/372#issuecomment-537965676!

**Comment by ifox** [comment]
Merged in afd14245. Thank you @ferpetrelli!

**Comment by ifox** [comment]
Merged in afd14245. Thank you @ferpetrelli!

**Pull request closed without merging**